lib/propshaft/server.rb

```diff
@@ -2,30 +2,39 @@
 require "rack/version"
 
 class Propshaft::Server
-  def initialize(assembly)
+  def initialize(app, assembly)
+    @app = app
     @assembly = assembly
   end
 
   def call(env)
     execute_cache_sweeper_if_updated
-    path, digest = extract_path_and_digest(env)
-
-    if (asset = @assembly.load_path.find(path)) && asset.fresh?(digest)
-      compiled_content = asset.compiled_content
-
-      [
-        200,
-        {
-          Rack::CONTENT_LENGTH  => compiled_content.length.to_s,
-          Rack::CONTENT_TYPE    => asset.content_type.to_s,
-          VARY                  => "Accept-Encoding",
-          Rack::ETAG            => "\"#{asset.digest}\"",
-          Rack::CACHE_CONTROL   => "public, max-age=31536000, immutable"
-        },
-        [ compiled_content ]
-      ]
+
+    path = env["PATH_INFO"]
+    method = env["REQUEST_METHOD"]
+
+    if (method == "GET" || method == "HEAD") && path.start_with?(@assembly.prefix)
```

Comment:
`HEAD` requests should return an empty body

https://github.com/rack/rack/blob/ee7ac5a1db5bc5c65e4b83342b8f4df88ef3c075/lib/rack/lint.rb#L819

Reply:
`Rack::Head` would take care of that, but yes, good point.